ui/components-react/src/components-react/filter-builder/FilterBuilderState.ts

```diff
@@ -167,34 +170,147 @@ export class PropertyFilterBuilderActions {
       rule.value = value;
     });
   }
+
+  /**
+   * Sets error messages of the rules specified by id.
+   * If rule id is not present in the map, then its error message will be cleared.
+   */
+  public setRuleErrorMessages(ruleIdsAndErrorMessages: Map<string, string>) {
+    this.updateState((state) => {
+      const setErrorMessages = (item: PropertyFilterBuilderRuleGroupItem) => {
+        if (isPropertyFilterBuilderRuleGroup(item)) {
+          item.items.forEach((itm) => {
+            setErrorMessages(itm);
+          });
+        } else {
+          item.errorMessage = ruleIdsAndErrorMessages.get(item.id);
+        }
+      };
+
+      setErrorMessages(state.rootGroup);
+    });
+  }
 }
 
 /**
  * Function to check if supplied [[PropertyFilterBuilderRuleGroupItem]] is [[PropertyFilterBuilderRuleGroup]].
- * @internal
+ * @beta
  */
 export function isPropertyFilterBuilderRuleGroup(
   item: PropertyFilterBuilderRuleGroupItem
 ): item is PropertyFilterBuilderRuleGroup {
   return (item as any).items !== undefined;
 }
 
+/**
+ * Props for [[usePropertyFilterBuilder]]
+ * @beta
+ */
+export interface UsePropertyFilterBuilderProps {
+  initialFilter?: PropertyFilter;
+}
 /**
  * Custom hook that creates state for [[PropertyFilterBuilder]] component. It creates empty state or initializes
  * state from supplied initial filter.
- * @internal
+ * @returns root group of the [[PropertyFilterBuilder]], actions for manipulating state and validate function, which is used for validating rules.
```

Comment:
I think it would be better to define interface for return type and document each part of it there.

Reply:
added